code/_helpers/heap.dm

```diff
@@ -0,0 +1,80 @@
+//////////////////////
+//datum/heap object
+//////////////////////
+
+/datum/heap
+	var/list/L
+	var/cmp
+
+/datum/heap/New(compare)
+	L = new()
+	cmp = compare
+
+/datum/heap/Destroy(force, ...)
+	for(var/i in L) // because this is before the list helpers are loaded
+		qdel(i)
+	L = null
+	return ..()
+
+/datum/heap/proc/is_empty()
+	return !length(L)
```

Comment:
Seems odd to have a proc for this given proc overhead and the simplicity of the check.